keras_nlp/utils/text_generation_test.py

```diff
@@ -113,6 +113,46 @@ def token_probability_fn(inputs):
         expected_outputs = tf.concat([inputs, expected_outputs], axis=1)
         self.assertAllEqual(outputs, expected_outputs)
 
+    def test_model_compile(self):
+        def token_probability_fn(inputs):
+            prob = tf.constant([[0.01, 0.01, 0.08, 0.9]])
+            return tf.repeat(prob, 2, axis=0)
+
+        max_length = 5
+
+        class TestModel(tf.keras.Model):
+            def call(self, inputs, training=False):
+                if not training:
+                    generated = greedy_search(
+                        token_probability_fn,
+                        inputs,
+                        max_length=max_length,
+                        end_token_id=2,
+                        pad_token_id=0,
+                    )
+                    return generated
+                else:
+                    return inputs
+
+        def dummy_metric(y_true, y_pred):
+            if tf.reduce_all(tf.equal(y_true, y_pred)):
+                return 1
+            else:
+                return 0
+
+        inputs = tf.constant([[0, 1], [1, 2]])
+        model = TestModel()
+        model.compile(metrics=[dummy_metric])
+
+        expected_outputs = tf.tile([[3], [0]], [1, max_length - 2])
+        expected_outputs = tf.concat([inputs, expected_outputs], axis=1)
+        self.assertEqual(
```

Comment:
would just using mode.predict work? that would still hit all the compiled function paths, and allow you to avoid all this dummy metric stuff, which is hard to read

Reply:
Could also be good to test the call on a batched dataset (where batch size not statically known), and on a single constant input, as you are doing here.